### ISSUE TITLE
Double ;; Induces Parser Error

### ISSUE THREAD

**Issue by betanalpha** [stan-dev/stanc3#469]
#### Description:

Extra semicolons ending a line throw a parser error despite the fact that an empty line causes no problems.

#### Reproducible Steps:

```
parameters {
  real x1;;
  real x2;
}

model {
  x1 ~ normal(0, 1);
  x2 ~ normal(0, 1);
}
```

#### Current Output:

```
SYNTAX ERROR, MESSAGE(S) FROM PARSER:

  error in 'bad.stan' at line 2, column 11
  -------------------------------------------------
     1: parameters {
     2:   real x1;;
                  ^
     3:   real x2;
  -------------------------------------------------

PARSER EXPECTED: <one of the following:
  a variable declaration, beginning with type,
      (int, real, vector, row_vector, matrix, unit_vector,
       simplex, ordered, positive_ordered,
       corr_matrix, cov_matrix,
       cholesky_corr, cholesky_cov
  or '}' to close variable declarations>
```

#### Expected Output:

No error or something more explicit like "Double semicolons not allowed".

#### Current Version:
v2.16.0

**Comment by mcol** [comment]
This model:
```
parameters {
  real x1;
  real x2;
}
transformed data {
  real x;
  x = 10; ;
}
model {
  x1 ~ normal(0, 1);
  x2 ~ normal(0, 1);
}
```
still triggers a confusing error message:
```
Syntax error in 'tmp.stan', line 1, column 0 to line 4, column 1, parsing error:
   -------------------------------------------------
     2:    real x1;
     3:    real x2;
     4:  }
          ^
     5:
     6:  transformed data {
   -------------------------------------------------

"transformed parameters {", "model {" or "generated quantities {" expected after end of parameters block.
```

**Comment by bob-carpenter** [comment]
The syntax is that a declaration block consists of a sequence of declarations.  Whitespace is always flexible in a Stan program---a single whitespace character is equivalent to any non-empty sequence of whitespace characters everywhere.

The reason a single semicolon isn't allowed in a declaration block is that it's not a declaration.  

Single semicolons are statements.  So they can be used on their own in blocks that allow general statements.

We could relax the requirement on declarations and allow multiple semicolons in declarations block.  Otherwise, the current error message is correct.  It could maybe be simplified to say that the parser expected a declaration and found a semicolon, though technically end-of-block brace is also OK in this context.

want to point out that whitespace lines with no variable declarations *are* allowed.  Whitepsace is always arbitrary, where one space character is equal to an arbitrary non-empty sequence of any whitespace characters (in ASCII).



 don't know why this popped up again, but I reread the response and want to point out that empty lines are allowed.  It's just that empty declarations are not allowed.  Empty statements are general statements, but the only statements allowed in declarations are declarations.

**Comment by bob-carpenter** [comment]
This program is legal.  General statements are allowed at the end of the transformed data block.

> transformed data {
>   real x;
>   x = 10; ;
> }

...

> still triggers a confusing error message:

**Comment by WardBrian** [comment]
This was closed by #988
